### PR TITLE
IP-Access-List Read and Profile Fix

### PR DESCRIPTION
**Pull request by Adelmar92**
## Description

the next updates were made for Project IP-Access-List:

- Removed API keys and Implemented Profile, with new Client in each method
- Fixed primary identifiers
- Fixed read

Link to any related issue(s): 

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: cfn-resources/project-ip-access-list/cmd/resource/resource.go [hunk 1]
```diff
@@ -48,19 +47,28 @@ func validateModel(fields []string, model *Model) *handler.ProgressEvent {
...
-	_, _ = logger.Debugf("currentModel: %+v, prevModel: %+v", currentModel, prevModel)
 
 	if errEvent := validateModel(CreateRequiredFields, currentModel); errEvent != nil {
 		return *errEvent, nil
 	}
 
-	_, _ = logger.Debugf("currentModel: %+v, prevModel: %+v", currentModel, prevModel)
+	if currentModel.Profile == nil {
+		currentModel.Profile = pointy.String(util.DefaultProfile)
+	}
```
Comment (colm-quinn) [inline cfn-resources/project-ip-access-list/cmd/resource/resource.go:57]: From another PR, I think we're using a different pattern for checking here. Ideally, lets standardise. See https://github.com/mongodb/mongodbatlas-cloudformation-resources/pull/326/files#r1113257393 for the example.

File: cfn-resources/project-ip-access-list/docs/README.md [hunk 2]
```diff
@@ -68,22 +60,24 @@ _Type_: <a href="listoptions.md">listOptions</a>
 
 _Update requires_: [No interruption](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-updating-stacks-update-behaviors.html#update-no-interrupt)
 
-## Return Values
+#### Profile
 
-### Ref
+Profile used to provide credentials information, (a secret with the cfn/atlas/profile/{Profile}, is required), if not provided default is used
```
Comment (colm-quinn) [inline cfn-resources/project-ip-access-list/docs/README.md:65]: Let's keep a standard text - from the encryption at rest PR, I think this works better:
```suggestion
The profile is defined in AWS Secret manager. See [Secret Manager Profile setup](../../../examples/profile-secret.yaml).
```

File: examples/project-ip-access-list/ip-access-list.yaml [hunk 3]
```diff
@@ -2,14 +2,10 @@
 AWSTemplateFormatVersion: '2010-09-09'
 Description: MongoDB Atlas AWS CloudFormation Quickstart for MongoDB::Atlas::ProjectIpAccessList
 Parameters:
-  PublicKey:
-    Description: "Your MongoDB Cloud Public API Key"
+  Profile:
+    Description: "MongoDB Atlas Proj Profile Name"
```
Comment (colm-quinn) [inline examples/project-ip-access-list/ip-access-list.yaml:6]:
```suggestion
    Description: "MongoDB Atlas Project Profile Name"
```

**Review by colm-quinn — COMMENTED** [review]
Looking good, some general comments on standardising across resources.

File: cfn-resources/project-ip-access-list/mongodb-atlas-projectipaccesslist.json [hunk 4]
```diff
@@ -95,35 +84,38 @@
...
   "readOnlyProperties": [
-    "/properties/Id",
-    "/properties/TotalCount"
+    "/properties/AccessList/CIDRBlock",
+    "/properties/AccessList/IPAddress",
+    "/properties/AccessList/AwsSecurityGroup"
+  ],
+  "createOnlyProperties": [
+    "/properties/ProjectId",
+    "/properties/Profile"
```
Comment (SuperMohit) [inline cfn-resources/project-ip-access-list/mongodb-atlas-projectipaccesslist.json:114]: Why Profile is set to be createOnly? Here is the definition of creation:
createOnlyProperties
Resource properties can be specified by the user only during resource creation.

Comment (SuperMohit) [reply cfn-resources/project-ip-access-list/mongodb-atlas-projectipaccesslist.json:114]: I guess this is again a hack to retrieve the Profile by CFN during Read/List?

Comment (andreaangiolillo) [reply cfn-resources/project-ip-access-list/mongodb-atlas-projectipaccesslist.json:114]: Shouldn't the profile be a primary identifier?

Comment (SuperMohit) [reply cfn-resources/project-ip-access-list/mongodb-atlas-projectipaccesslist.json:114]: Sorry, maybe wrong question. I am a bit confused about these attributes.

Comment (andreaangiolillo) [reply cfn-resources/project-ip-access-list/mongodb-atlas-projectipaccesslist.json:114]: it makes sense. Thanks!

**Review by andreaangiolillo — APPROVED** [review]
LGTM

File: cfn-resources/project-ip-access-list/mongodb-atlas-projectipaccesslist.json (hunk 4 above)
Comment (SuperMohit) [reply cfn-resources/project-ip-access-list/mongodb-atlas-projectipaccesslist.json:114]: Thanks for the explanation.

File: cfn-resources/project-ip-access-list/cmd/resource/resource.go [hunk 5]
```diff
@@ -48,19 +47,28 @@ func validateModel(fields []string, model *Model) *handler.ProgressEvent {
...
 	setup()
-	_, _ = logger.Debugf("currentModel: %+v, prevModel: %+v", currentModel, prevModel)
 
 	if errEvent := validateModel(CreateRequiredFields, currentModel); errEvent != nil {
 		return *errEvent, nil
 	}
 
-	_, _ = logger.Debugf("currentModel: %+v, prevModel: %+v", currentModel, prevModel)
+	if currentModel.Profile == nil || *currentModel.Profile == "" {
+		currentModel.Profile = pointy.String(util.DefaultProfile)
```
Comment (SuperMohit) [inline cfn-resources/project-ip-access-list/cmd/resource/resource.go:56]: just for consistency.
```suggestion
		currentModel.Profile = aws.String(util.DefaultProfile)
```

**Review by SuperMohit — APPROVED** [review]

**Review by andreaangiolillo — APPROVED** [review]

**Pull request merged**